### PR TITLE
Some suggested changes

### PR DESCRIPTION
**Pull request by MCGallaspy**
This PR both demonstrates using a non-callable `hooks` attribute for plugins, as well as adding quite a bit of documentation.

### CODE REVIEW

File: kolibri/core/kolibri_plugin.py [hunk 1]
```diff
@@ -14,15 +14,22 @@
...
-    def hooks(self):
-        return {
-            NAVIGATION_POPULATE: self.main_navigation
-        }
+class KolibriCore(KolibriPluginBase):
+    """
+    The most minimal plugin possible. Because it's in the core, it doesn't define ``enable`` or ``disable``. Those
+    methods should never be called for this plugin.
+    """
+    hooks = {
```
Comment (benjaoming) [inline kolibri/core/kolibri_plugin.py:33]: I have to disagree with this version... if it's a callable, it gives the plugin the power to do stuff more conditionally. Like defining which hooks it's going to handle and which callables it wants to give them... this assignment is static and happens and load time.

File: kolibri/core/kolibri_plugin.py [hunk 2]
```diff
@@ -14,15 +14,22 @@
...
+        NAVIGATION_POPULATE: main_navigation
```
Comment (benjaoming) [inline kolibri/core/kolibri_plugin.py:34]: This callable is now statically defined, wouldn't that mean that the `ExtendedPlugin` inheritor further below won't be able to override this static reference?

File: kolibri/core/kolibri_plugin.py (hunk 1 above)
Comment (MCGallaspy) [inline kolibri/core/kolibri_plugin.py:33]: In particularly simple cases there's less boilerplate -- but perhaps you're right that more generally we'll want to do more with hooks than a load-time definition allows

File: kolibri/core/kolibri_plugin.py (hunk 2 above)
Comment (MCGallaspy) [inline kolibri/core/kolibri_plugin.py:34]: In this case yes, but below hooks is actually a property, so no.

**Comment by benjaoming** [comment]
I'm gonna merge this, but feel free to work more on the it during this Friday, but anything you have, please make the PR before signing off  for the weekend as I'll be working on it first thing Monday :)

**Comment by MCGallaspy** [comment]
I think we're in agreement about the attribute vs callable idea, i.e. callable is better. Not sure if I'll get a chance to work on it today as we're in the bay area for the day for a meeting.

**Comment by benjaoming** [comment]
I think we have the same understanding, too. Great docstrings btw!

**Pull request merged**